File: src/main/java/nl/hanze/week1/assignment1/observers/IBMObserver.java
package nl.hanze.week1.assignment1.observers;

import nl.hanze.week1.assignment1.Observer;
import nl.hanze.week1.assignment1.Stock;
import nl.hanze.week1.assignment1.StockGrabber;
import nl.hanze.week1.assignment1.View;

/**
 * Created by Laurens on 25-9-2017.
 */
public class IBMObserver extends BaseObserver implements Observer {

    private double ibmPrice;
    private StockGrabber stockGrabber;
    private View view;

    public IBMObserver(StockGrabber stockGrabber) {
        stockGrabber.register(this);
        this.stockGrabber = stockGrabber;
    }

    @Override
    public void update(double ibmPrice, double aaplPrice, double googPrice) {
        this.ibmPrice = ibmPrice;
        updateViews(Stock.IBM, ibmPrice);
    }
}

File: src/main/java/nl/hanze/assignment1/observers/IBMObserver.kt
package nl.hanze.assignment1.observers

import nl.hanze.assignment1.*

/**
 * Created by Laurens on 25-9-2017.
 */
class IBMObserver(private val stockGrabber: Subject) : BaseObserver(), Observer {

    private var ibmPrice: Double = 0.0

    init {
        stockGrabber.register(this)
    }

    override fun update(stockUpdate: StockUpdate) {
        if (stockUpdate.stock == Stock.IBM) {
            this.ibmPrice = stockUpdate.price
            updateViews(stockUpdate)
        }
    }
}
